File: app/src/main/java/project/android/footstamp/adapter/PagerAdapter.java
package project.android.footstamp.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import project.android.footstamp.fragment.ViewFragment;
import project.android.footstamp.fragment.ViewFragment2;

public class PagerAdapter extends FragmentStatePagerAdapter {
    int mNumOfTabs;

    public PagerAdapter(@NonNull FragmentManager fm, int NumOfTabs) {
        super(fm, NumOfTabs);
        this.mNumOfTabs = NumOfTabs;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position){
            case 0: return new ViewFragment();
            case 1: return new ViewFragment2();
            default: return null;
        }
    }

    @Override
    public int getCount() {
        return mNumOfTabs;
    }
}

File: app/src/main/java/project/android/footstamp/adapter/PagerAdapter.kt
package project.android.footstamp.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import project.android.footstamp.fragment.GalleryFragment
import project.android.footstamp.fragment.ViewFragment
import project.android.footstamp.fragment.ViewFragment2

class PagerAdapter(fa:FragmentActivity) : FragmentStateAdapter(fa){
    var fragments = listOf<Fragment>()
    override fun getItemCount(): Int {
        return fragments.size
    }

    override fun createFragment(position: Int): Fragment {
        return fragments.get(position)
    }

}